cmd/flux/cmd/test.go

```diff
@@ -362,3 +344,55 @@ func (t *TestReporter) Summarize(tests []*Test) {
 	}
 	fmt.Printf("\n---\nRan %d tests with %d failure(s)\n", len(tests), failures)
 }
+
+type TestSetupFunc func(ctx context.Context) (TestExecutor, error)
+
+type TestExecutor interface {
+	Run(pkg *ast.Package) error
+	io.Closer
+}
+
+func NewTestExecutor(ctx context.Context) (TestExecutor, error) {
+	return testExecutor{}, nil
+}
+
+type testExecutor struct{}
+
+func (testExecutor) Run(pkg *ast.Package) error {
+	jsonAST, err := json.Marshal(pkg)
+	if err != nil {
+		return err
+	}
+	c := lang.ASTCompiler{AST: jsonAST}
+
+	ctx := executetest.NewTestExecuteDependencies().Inject(context.Background())
+	program, err := c.Compile(ctx, runtime.Default)
+	if err != nil {
+		return errors.Wrap(err, codes.Invalid, "failed to compile")
+	}
+
+	alloc := &memory.Allocator{}
+	query, err := program.Start(ctx, alloc)
+	if err != nil {
+		return errors.Wrap(err, codes.Inherit, "error while executing program")
+	}
+	defer query.Done()
+
+	results := flux.NewResultIteratorFromQuery(query)
+	for results.More() {
+		result := results.Next()
+		err := result.Tables().Do(func(tbl flux.Table) error {
+			// The data returned here is the result of `testing.diff`, so any result means that
+			// a comparison of two tables showed inequality. Capture that inequality as part of the error.
+			// XXX: rockstar (08 Dec 2020) - This could use some ergonomic work, as the diff output
+			// is not exactly "human readable."
+			return fmt.Errorf("%s", table.Stringify(tbl))
+		})
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (testExecutor) Close() error { return nil }
```

Comment:
Is this complete necessary?

Reply:
I added this for influxdb and idpe since they would open resources and need to close them. But it largely depends if the test command is responsible for standing up infrastructure or not.